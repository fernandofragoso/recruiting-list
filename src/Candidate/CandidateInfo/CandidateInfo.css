.CandidateInfo__item {
  line-height: 30px;
}

.CandidateInfo__pill {
  border: 1px solid #000;
  border-radius: 15px;
  font-size: 12px;
  padding: 5px 10px;
}

.CandidateInfo__pill--new {
  color: #E9544E;
  border: 1px solid #E9544E;
  margin: 0 10px;
  padding: 2px 10px;
}

.CandidateInfo__pill--skill {
  border: 1px solid #E5E5E5;
  background-color: #F1F2F4;
  margin: 0 2px;
}