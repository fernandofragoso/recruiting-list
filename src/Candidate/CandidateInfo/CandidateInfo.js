import React, { Component } from 'react';
import './CandidateInfo.css';

import workicon from '../../img/work-icn.svg';
import studyicon from '../../img/study-icn.svg';
import locationicon from '../../img/location-icn.svg';
import skillsicon from '../../img/skills-icn.svg';

class CandidateInfo extends Component {
  
  constructor(props) {
    super(props);
  }
  
  render() {

    let newPill = null;
    if (this.props.candidate.newThisWeek) {
      newPill = <span className="CandidateInfo__pill CandidateInfo__pill--new">Novo essa semana</span>
    }

    return (
      <div className="CandidateInfo">
        <h3>{this.props.candidate.name} {newPill}</h3>
        <h4>{this.props.candidate.career}</h4>
        <div className="CandidateInfo__item">
          <img className="icon" alt="Work icon" src={workicon} />
          {this.props.candidate.lastCompanies.join(', ')}
        </div>
        <div className="CandidateInfo__item">
          <img className="icon" alt="Study icon" src={studyicon} />
            {this.props.candidate.university}
          </div>
        <div className="CandidateInfo__item">
          <img className="icon" alt="Location icon" src={locationicon} /><span>Disposto a trabalhar em: </span>
          {this._getCities(this.props.candidate.cities)}
        </div>
        <div className="CandidateInfo__item">
          <img className="icon" alt="Skill icon" src={skillsicon} /><span>Principais habilidades: </span>
          {this.props.candidate.mainSkills.map((skill, index) => {
            return <span key={index} className="CandidateInfo__pill CandidateInfo__pill--skill">
              {skill}
            </span>
          })}
        </div>
      </div>
    );
  }

  _getCities(cities) {
    if (cities.length > 3) {
      return <span>
        {cities.slice(0,2).join(", ")} e mais <strong>{cities.length - 2} cidades</strong>
      </span>
    } else {
      return cities.join(", ");
    }
  }
}

export default CandidateInfo;
