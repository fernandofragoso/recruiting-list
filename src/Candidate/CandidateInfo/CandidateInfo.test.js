import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CandidateInfo from './CandidateInfo';

const getCandidate = (candidate) => {
  return <CandidateInfo candidate={candidate} />;
}

const candidate1 = {
  "id": 3,
  "imageUrl":
    "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTFWfkA6fPozy8qlbL8q282B2jWo10s-UK2bYFLv1b_v32r5TAF",
  "name": "Raoul Marquardt",
  "newThisWeek": true,
  "career": "Chief Security Representative",
  "lastCompanies": [
    "Thompson, Hintz and Walker",
    "Aufderhar, Goldner and Dach",
    "Predovic, Gutkowski and Lubowitz"
  ],
  "university": "Kentucky",
  "cities": ["Lake Bridgette", "Abeton"],
  "mainSkills": ["Creative", "Solutions"]
}

const candidate2 = {
  "id": 3,
  "imageUrl":
    "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTFWfkA6fPozy8qlbL8q282B2jWo10s-UK2bYFLv1b_v32r5TAF",
  "name": "Raoul Marquardt",
  "newThisWeek": false,
  "career": "Chief Security Representative",
  "lastCompanies": [
    "Thompson, Hintz and Walker",
    "Aufderhar, Goldner and Dach",
    "Predovic, Gutkowski and Lubowitz"
  ],
  "university": "Kentucky",
  "cities": ["Lake Bridgette", "Abeton", "Rio de Janeiro", "Recife"],
  "mainSkills": ["Creative", "Solutions"]
}

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CandidateInfo candidate={candidate1} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly with less than 3 cities and new this week', () => {
  const tree = renderer.create(getCandidate(candidate1).toJSON);
  expect(tree).toMatchSnapshot();
});

it('renders correctly with more than 3 cities', () => {
  const tree = renderer.create(getCandidate(candidate2).toJSON);
  expect(tree).toMatchSnapshot();
});