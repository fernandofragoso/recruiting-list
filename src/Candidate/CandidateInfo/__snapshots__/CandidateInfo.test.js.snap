// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with less than 3 cities and new this week 1`] = `null`;

exports[`renders correctly with more than 3 cities 1`] = `null`;
